code/_globalvars/misc.dm

```diff
@@ -20,3 +20,7 @@ var/map_ready = 0
 	However, using lots of spawn() has a severe performance impact, and often results in spaghetti-code
 	map_ready will be set to 1 when world/New() is called (which happens just after the map is instanced)
 */
+
+var/global/current_year = null
+var/global/current_month = null
+var/global/current_day = null

```

Comment:
The idea was to reduce processing by having the current date only need to be set once.